terraform/modules/instance/main.tf

```diff
@@ -0,0 +1,25 @@
+
+data "terraform_remote_state" "network" {
+  backend = "s3"
```

Comment:
Here you locking down your module to tf remote state.
In makes impossible to use it outside of it's (remote state "network") scope.
It's obivious candidate to be replaced with input variable.